From 4e8fff181b343f1d76190566f8879365c90526b3 Mon Sep 17 00:00:00 2001
From: mickhuang <mickhuang@ami.com>
Date: Wed, 4 Sep 2024 17:10:38 +0800
Subject: [PATCH] Enable USB Port B as gadget device

- Disable EHCI and UHCI node in dts.
- Enable UDC node in dts.

Signed-off-by: mickhuang <mickhuang@ami.com>
---
 arch/arm/boot/dts/aspeed-ast2600-evb.dts | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/aspeed-ast2600-evb.dts b/arch/arm/boot/dts/aspeed-ast2600-evb.dts
index 2ac8ef66e10e..e0b611eeff99 100644
--- a/arch/arm/boot/dts/aspeed-ast2600-evb.dts
+++ b/arch/arm/boot/dts/aspeed-ast2600-evb.dts
@@ -560,11 +560,17 @@ &fsim0 {
 };
 
 &ehci1 {
-	status = "okay";
+	status = "disabled";
 };
 
 &uhci {
-	status = "okay";
+	status = "disabled";
+};
+
+&udc {
+        status = "okay";
+        pinctrl-names = "default";
+        pinctrl-0 = <&pinctrl_usb2bd_default>;
 };
 
 &sdc {
-- 
2.34.1

